cmd/influx_inspect/export/parquet/models/points.go

```diff
@@ -0,0 +1,389 @@
+	// of tags.
+	var i int
+	walkTags(buf, func(key, value []byte) bool {
+		dst[i].Key, dst[i].Value = key, value
+		i++
+		return true
+	})
+	return dst[:i]
+}
+
+// Escaped represents an escaped portion of line-protocol syntax
+// using an underlying byte slice. It's defined as a struct to avoid
+// unintentional implicit conversion between []byte and Escaped.
+//
+// In general you should avoid using the B field unless necessary,
+// as doing so risks breaking the typed assertion that the data is
+// validly escaped syntax.
+type Escaped struct{ B []byte }
+
+// MakeEscaped returns the equivalent of Escaped{B: b}.
+// Only use this when you're sure that b is either already
+// escaped or (in tests) when b has no characters that need escaping.
+func MakeEscaped(b []byte) Escaped {
+	return Escaped{b}
+}
+
+// S returns the EscapedString form of m.
+func (e Escaped) S() EscapedString {
+	return EscapedString(e.B)
+}
+
+// Clone makes a copy of e and its underlying byte slice.
+func (e Escaped) Clone() Escaped {
+	return Escaped{B: cloneBytes(e.B)}
+}
+
+// Slice returns a slice of the underlying data from i0 to i1.
+func (e Escaped) Slice(i0, i1 int) Escaped {
+	return Escaped{e.B[i0:i1]}
+}
+
+// SliceToEnd returns a slice of the underlying data from i0 to the end of the slice.
+func (e Escaped) SliceToEnd(i0 int) Escaped {
+	return e.Slice(i0, len(e.B))
+}
+
+// EscapedString represents an escaped portion of line-protocol syntax.
+// See Escaped for more information.
+type EscapedString string
+
+// B returns the byte-slice form of e.
+func (e EscapedString) B() Escaped {
+	return Escaped{[]byte(e)}
+}
+
+func cloneBytes(b []byte) []byte {
+	return append([]byte(nil), b...)
+}
+
+func UnescapeToken(in Escaped) []byte {
+	out := in.B
+	if bytes.IndexByte(out, '\\') == -1 {
+		return out
+	}
+
+	for i := range tagEscapeCodes {
+		c := &tagEscapeCodes[i]
+		if bytes.IndexByte(out, c.k[0]) != -1 {
+			out = bytes.Replace(out, c.esc[:], c.k[:], -1)
+		}
+	}
+	return out
+}
+
+func UnescapeMeasurement(inm Escaped) []byte {
+	in := inm.B
+	if bytes.IndexByte(in, '\\') == -1 {
+		return in
+	}
+
+	for i := range measurementEscapeCodes {
+		c := &measurementEscapeCodes[i]
+		if bytes.IndexByte(in, c.k[0]) != -1 {
+			in = bytes.Replace(in, c.esc[:], c.k[:], -1)
+		}
+	}
+	return in
+}
+
+// The following constants allow us to specify which state to move to
+// next, when scanning sections of a Point.
+const (
+	tagKeyState = iota
+	tagValueState
+	fieldsState
+)
+
+// scanMeasurement examines the measurement part of a Point, returning
+// the next state to move to, and the current location in the buffer.
+func scanMeasurement(bufe Escaped, i int) (int, int, error) {
+	buf := bufe.B
+	// Check first byte of measurement, anything except a comma is fine.
+	// It can't be a space, since whitespace is stripped prior to this
+	// function call.
+	if i >= len(buf) || buf[i] == ',' {
+		return -1, i, fmt.Errorf("missing measurement")
```

Comment:
Can we print the buffer inside quotes with `%q` in the error message to help debugging?